src/masonite/tests/TestCase.py

```diff
@@ -25,6 +27,18 @@ def setUp(self):
             self.startTestRun()
         self.withoutCsrf()
         self._exception_handling = False
+        # boot providers as they won't not be loaded if the test is not doing a request
+        self.application.bind("environ", generate_wsgi())
+        try:
+            for provider in self.application.get_providers():
+                # if no request is made we don't need RouteProvider, and we can't load it anyway
+                # because we don't have created a CSRF token yet
+                if not isinstance(provider, RouteProvider):
```

Comment:
Also i thought we get around the csrf checks by setting a cookie and the csrf input?

Reply:
hmm not sure what the best approach is but if this works i'll merge it in and we can revisit it later